apps/server/lib/lexical/server/code_intelligence/completion.ex

```diff
@@ -61,6 +62,15 @@ defmodule Lexical.Server.CodeIntelligence.Completion do
       Enum.empty?(prefix_tokens) or not context_will_give_meaningful_completions?(env) ->
         Completion.List.new(items: [], is_incomplete: true)
 
+      Env.in_context?(env, :struct_arguments) and not Env.in_context?(env, :value) and
+          not prefix_is_trigger?(env) ->
+        {line, column} = Env.prefix_alias_position(env)
+        poisiton = Position.new(line, column)
+
+        project
+        |> RemoteControl.Api.complete_struct_fields(env.document, poisiton)
+        |> StructField.translate(Env, env)
```

Comment:
this is calling a protocol directly, which isn't how protocols are supposed to be used. I suspect it's necessary because you're calling a different function for struct completions rather than "normal" completions, which feels like it's incorrect to me. Why was that necessary? 

The feeling I had is that you should ask `remote_control` for the completions and that `remote_control` should deliver struct field completions back. You shouldn't have to do a check prior to calling it and call a different function.

Reply:
Ok, I changed this.